pkg/apis/cluster/v1alpha1/common_types.go

```diff
@@ -39,5 +42,17 @@ type ProviderConfig struct {
 // ProviderConfigSource represents a source for the provider-specific
 // resource configuration.
 type ProviderConfigSource struct {
-	// TODO(roberthbailey): Fill these in later
+	// The machine class from which the provider config should be sourced.
+	// +optional
+	MachineClass *MachineClassRef `json:machineClass,omitempty`
```

Comment:
I just want to point out I believe this is different from the way StorageClass and RuntimeClass handles ref.  I guess it's ok since we're not handling the class the same way as them.  They use just a string.  Their controller matches the class' string name to match up the provider.  So if we were following the same pattern, we would have some controller that matched up the string with a provider known to the controller.  It's been a few months since I looked at this, but I believe that is the pattern.